### DISCUSSION TITLE
New subproject: Theengs Explorer?

### DISCUSSION THREAD

**Discussion by koenvervloesem** [theengs/decoder#132] [Show and tell]
Hi, I created a first proof-of-concept of something I call Theengs Explorer: a text user interface built with [Textual](https://github.com/Textualize/textual) for Theengs Decoder. It continually scans for BLE advertisements and for each device shows the device information, detected advertisement interval, decoded data and the last advertisement data. The list is scrollable, and it's great to explore the BLE devices around you.

I have to clean up the code before publishing it, but here's a screenshot:

![Schermafdruk van 2022-06-26 22-59-56](https://user-images.githubusercontent.com/214673/175833648-f1416e7e-b9e1-4086-9ab3-4e10d94cf555.png)

Is this an interesting project to build in the theengs organization, or is this out of scope and more something for a third-party project?

**Comment by DigiH** [comment]
Hi @koenvervloesem,

while I can't say much about this possibly being under the Theengs umbrella, really @1technophile's decision, I like what I see on your screenshot 😃 

Can't wait to try it out once you have cleaned up the code.

Thanks

**Comment by 1technophile** [comment]
Hi @koenvervloesem,

That's very cool!
We will be happy to welcome this project. Let me open a repository to welcome it.

**Comment by 1technophile** [comment]
Repository created https://github.com/theengs/explorer

**Reply by koenvervloesem** [reply]
Thanks! I'll upload my code in a few days.

**Comment by koenvervloesem** [comment]
I've published the initial code on https://github.com/theengs/explorer. Enjoy!

**Reply by DigiH** [reply]
While I've already installed textual in anticipation ;) TheengsExplorer doesn't seem to available yet for an official pip3 install.

Is it possible for you @koenvervloesem or @1technophile to activate the Discussions on theengs/explorer? Might be easier and more suitable for addressing some things, instead of always opening issues.

Like TheengsExplorer including the companies list and decoding it for display. Which is a great thing for when the manufacturerdata actually adheres to this supposedly mandatory convention, but unfortunately many devices don't and then you get varying changing companies displayed if such a device, for example, uses these bytes to send their temperature.

Current new decoder discussion, where the actual device temperature is displayed as the compan(y|ies) by the nrfConnect app

https://github.com/theengs/decoder/discussions/131

While we also have already discussed this option/problem for Theengs Decoder, and decided against it for the above reason, I still think it might be an option to add a flag to all device decoders, as we pretty much instantly know whenever a new decoder is developed, if this issue is present or if the device actually adheres to the company id convention. This would make it possible for any app using the Decoder library, like TheengsExplorer, to correctly display it when available, but also to ignore it when it would only produces nonsensical results.

**Reply by koenvervloesem** [reply]
Good idea, I've set up Discussions in the explorer repository.

**Reply by 1technophile** [reply]
> While we also have already discussed this option/problem for Theengs Decoder, and decided against it for the above reason, I still think it might be an option to add a flag to all device decoders, as we pretty much instantly know whenever a new decoder is developed, if this issue is present or if the device actually adheres to the company id convention

I agree, we could have a flag for the devices that doesn't use properly the manufacturer id, as they are the exception. And let the other ones.

**Comment by DigiH** [comment]
@koenvervloesem - might be nice to have the above screenshot, or any later up to date one, on the Theengs Explorer README project page as an immediate visual description :)

**Reply by koenvervloesem** [reply]
Good point, I added a more recent screenshot.
